src/modules/worlds/reducer.ts

```diff
@@ -59,6 +117,107 @@ export function worldsReducer(state: WorldsState = INITIAL_STATE, action: Worlds
         }
       }
     }
+    case GET_WORLD_PERMISSIONS_SUCCESS: {
+      const { worldName, permissions } = action.payload
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        worldsPermissions: {
+          ...state.worldsPermissions,
+          [worldName]: permissions
+        }
+      }
+    }
+    case POST_WORLD_PERMISSIONS_SUCCESS: {
+      const { worldName, worldPermissionNames, worldPermissionType } = action.payload
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        worldsPermissions: {
+          ...state.worldsPermissions,
+          [worldName]: {
+            ...state.worldsPermissions[worldName],
+            [worldPermissionNames]: {
+              type: worldPermissionType,
+              ...(worldPermissionType === WorldPermissionType.AllowList && { wallets: [] })
+            }
+          }
+        }
+      }
+    }
+    case PUT_WORLD_PERMISSIONS_SUCCESS: {
+      const { worldName, worldPermissionNames, worldPermissionType, newData } = action.payload
+
+      let worldPermissions: WorldPermissions = state.worldsPermissions[worldName]
+
+      let newArrayWallet: string[] = []
+      if ((worldPermissions[worldPermissionNames] as AllowListPermissionSetting).wallets) {
+        newArrayWallet = (worldPermissions[worldPermissionNames] as AllowListPermissionSetting).wallets
+      }
+      newArrayWallet.push(newData)
+      worldPermissions = {
+        ...worldPermissions,
+        [worldPermissionNames]: {
+          type: worldPermissionType,
+          wallets: newArrayWallet
+        }
+      }
+
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        worldsPermissions: {
+          ...state.worldsPermissions,
+          [worldName]: {
+            ...state.worldsPermissions[worldName],
+            ...worldPermissions
+          }
+        }
+      }
+    }
+    case DELETE_WORLD_PERMISSIONS_SUCCESS: {
+      const { worldName, worldPermissionNames, worldPermissionType, address } = action.payload
+
+      let worldPermissions: WorldPermissions = state.worldsPermissions[worldName]
+
+      const newArrayWallet: string[] = (worldPermissions[worldPermissionNames] as AllowListPermissionSetting).wallets.filter(
+        wallet => wallet !== address
+      )
+
+      worldPermissions = {
+        ...worldPermissions,
+        [worldPermissionNames]: {
+          type: worldPermissionType,
+          wallets: newArrayWallet
+        }
+      }
+
+      return {
+        ...state,
+        loading: loadingReducer(state.loading, action),
+        worldsPermissions: {
+          ...state.worldsPermissions,
+          [worldName]: {
+            ...state.worldsPermissions[worldName],
```

Comment:
Isn't this already included in the `worldPermissions` variable?

Reply:
nop, the POST changes the type if the permission. For this cases we accept only unrestricted and allow-list, but in the future we could add also ntf and secret.